File: WinPaletter/GUI/Elements/Labels/LinkLabel.vb
Imports System.ComponentModel

Namespace UI.WP

    <Description("LinkLabel but with a proper hand cursor")>
    Public Class LinkLabel : Inherits Windows.Forms.LinkLabel

        Const WM_SETCURSOR As Integer = 32, IDC_HAND As Integer = 32649

        Protected Overrides Sub WndProc(ByRef m As Message)
            If m.Msg = WM_SETCURSOR Then
                Dim cursor As Integer = NativeMethods.User32.LoadCursor(0, IDC_HAND)
                NativeMethods.User32.SetCursor(cursor)
                m.Result = IntPtr.Zero
                Return
            End If

            MyBase.WndProc(m)
        End Sub

    End Class

End Namespace

File: WinPaletter/GUI/Elements/Labels/LinkLabel.cs
using System;
using System.ComponentModel;
using System.Windows.Forms;
using WinPaletter.NativeMethods;

namespace WinPaletter.UI.WP
{
    [Description("LinkLabel but with a proper hand cursor")]
    public class LinkLabel : System.Windows.Forms.LinkLabel
    {
        private const int IDC_HAND = 32649;

        protected override void WndProc(ref Message m)
        {
            if (m.Msg == (int)User32.WindowsMessage.SetCursor)
            {
                int cursor = User32.LoadCursor(0, IDC_HAND);
                User32.SetCursor(cursor);
                m.Result = IntPtr.Zero;
                return;
            }

            base.WndProc(ref m);
        }
    }
}